### PR TITLE
PEP 001 - Initial Porter Enhancement Proposal Process

### PR DESCRIPTION
**Pull request by carolynvs**
Thank you for submitting a proposal to the Porter project! Here is a quick checklist to
ensure that your proposal is ready to submit:

- [ ] Is your pull request named: `PEP NNN - <TITLE>`?
- [ ] Have you posted your proposal in the [forum](https://porter.sh/forum)?
- [ ] Have you gotten approval and a PEP number from a maintainer?
- [ ] Did you use the [PEP Template](https://github.com/getporter/proposals/blob/master/pep/000-PEP-TEMPLATE.md)?

### Forum Suggestion Link

https://github.com/getporter/porter/discussions/1344

### CODE REVIEW

File: pep/001-initial-pep-process.md
Comment (vdice) [inline pep/001-initial-pep-process.md:355]: Confused by "...the Porter repository updated to link to this repository"?

File: pep/001-initial-pep-process.md
Comment (vdice) [inline pep/001-initial-pep-process.md:359]: I'm confused by these next sections.  Are they the PEP-mandated sections for this particular 001 PEP?

File: pep/001-initial-pep-process.md
Comment (vdice) [inline pep/001-initial-pep-process.md:378]: Should we link to the GH issue here?  Or is that not a requirement for Open Issues listings in PEPs?

File: pep/001-initial-pep-process.md
Comment (vdice) [inline pep/001-initial-pep-process.md:249]: Do we want to formalize what each issue listing should look like?  Always be a link to a corresponding GH issue ... or can be more informal?

File: pep/001-initial-pep-process.md
Comment (carolynvs) [reply pep/001-initial-pep-process.md:268]: Before we merge the PR that status is **proposed**. We may not merge it and it could go nowhere. Once it is merged the status is **provisional**. 

I didn't want to have them submit it as proposed in the md, and then after merging have to go back and update the status to provisional. So I thought open pull requests for a new PEP are proposed, and it's not provisional until it's merged. Let me know if you have suggestions to make that less confusing!

File: pep/001-initial-pep-process.md
Comment (carolynvs) [reply pep/001-initial-pep-process.md:355]: I'll tweak that. I mean that multiple files in the porter repo need updates to tell people about PEPs.

File: pep/001-initial-pep-process.md
Comment (carolynvs) [reply pep/001-initial-pep-process.md:359]: I am doing the proposal for a PEP as a PEP. K8s and Helm did the same, essentially testing out the PEP process and using the proposal to have PEPs as the first PEP.

So since part of our PEP requirements is backwards compatability, security, etc you will see those sections filled out as well. The PEP process is outlined in its entirety in the specification section. That is what I will put in the contributing guide in the proposals repository when we agree that the PEP is implementable. The rest of the sections here represent the PEP process as outlined and will just stay in this proposal file.

File: pep/001-initial-pep-process.md
Comment (carolynvs) [reply pep/001-initial-pep-process.md:378]: Let me rename this section to "Open Concerns" or "Open Questions". These are not meant to be issues in GitHub. Just things that the Owner of the PEP has identified, and they are thinking about/discussing as they talk about the PEP with people and are building consensus. Most likely the conversations about this will be happening in the forum.

I can encourage people to put a link if there is an established discussion but that may not be the case most of the time.

File: pep/001-initial-pep-process.md
Comment (carolynvs) [reply pep/001-initial-pep-process.md:249]: I don't really want to formalize how people discuss the PEP. If we can, my preference is to keep this lightweight and only add requirements for things that would cause problems if unspecified.

File: pep/001-initial-pep-process.md
Comment (carolynvs) [reply pep/001-initial-pep-process.md:359]: I've added a horizontal rule to help call out where the specification ends.

**Comment by carolynvs** [comment]
@vdice I've incorporated your fixes, thanks for the editing! I've also tried to address some of the questions you had, can you take a look at the updated version and let me know if it's more clear?

**Review by vdice — APPROVED** [review]
Thanks for the clarifications!  LGTM.

File: pep/001-initial-pep-process.md
Comment (iennae) [inline pep/001-initial-pep-process.md:274]: is this supposed to be provisional and provisionable? these have the same root but they don't mean the same thing to me?

File: pep/001-initial-pep-process.md
Comment (iennae) [reply pep/001-initial-pep-process.md:270]: I think a drawing at the top of this to show states would be super helpful. The states make sense to me but it can be challenging to explain with words.

File: pep/001-initial-pep-process.md
Comment (iennae) [inline pep/001-initial-pep-process.md:352]: Should there be a **RETIRED** state? for when something goes through the whole implementation process and then later it's removed? It could be possible that it's not that something should be replaced but it's no longer needed/necessary for whatever reason. 

Just thinking through the process, would a new PEP that removes something be "IMPLEMENTED" and then the old one would be"REPLACED" ?

**Comment by iennae** [comment]
Overall, I think this looks great! It's a great example to implement and test the process out. Just need the clarification about the word provisional and provisionable really. I think adding the diagram is useful to.

File: pep/001-initial-pep-process.md
Comment (carolynvs) [reply pep/001-initial-pep-process.md:274]: Sorry no that's just my inability to spell.

File: pep/001-initial-pep-process.md
Comment (carolynvs) [reply pep/001-initial-pep-process.md:352]: In the process that I described, deprecation/removal of something falls under "implementation". Implemented means the PEP as described as implemented. So if I had feature X, the first PEP would suggest adding feature X and eventually be marked implemented. Later when we changed our minds, a new PEP to remove the feature is proposed and once implemented (by removing the feature), the original PEP marked as replaced.

I can add a comment clarifying that it is for all changes, including removal of existing functionality.

File: pep/001-initial-pep-process.md
Comment (iennae) [reply pep/001-initial-pep-process.md:275]: I could totally see it being intentional too.. that's why I was asking.

**Review by iennae — APPROVED** [review]
LGTM!

**Pull request merged**
